src/shapes/polygon.class.ts

```diff
@@ -1,12 +1,48 @@
 import { fabric } from '../../HEADER';
+import { halfPI, twoMathPi } from '../constants';
+import { Point } from '../point.class';
 import { TClassProperties } from '../typedefs';
+import { cos } from '../util/misc/cos';
+import { sin } from '../util/misc/sin';
 import { classRegistry } from '../util/class_registry';
 import { Polyline, polylineDefaultValues } from './polyline.class';
 
 export class Polygon extends Polyline {
   protected isOpen() {
     return false;
   }
+
+  static getRegularPolygonPoints({
+    numVertexes,
+    radius,
+  }: {
+    numVertexes: number;
+    radius: number;
+  }) {
+    const interiorAngle = twoMathPi / numVertexes;
+    // rotate the polygon by 1/2 the interior angle so that the polygon always has a flat side on the bottom
+    const rotationAdjustment =
+      -halfPI + (numVertexes % 2 === 0 ? interiorAngle / 2 : 0);
+    return new Array(numVertexes).fill(0).map((_, i) => {
+      const rad = i * interiorAngle + rotationAdjustment;
+      return new Point(cos(rad), sin(rad)).scalarMultiply(radius);
+    });
+  }
+
+  static createRegularPolygon({
+    numVertexes,
+    radius,
+    ...options
+  }: {
+    numVertexes: number;
+    radius: number;
+  } & Record<string, unknown>) {
+    return new this(this.getRegularPolygonPoints({ numVertexes, radius }), {
+      left: 0,
+      top: 0,
+      ...options,
+    });
+  }
```

Comment:
this extra static method seems like not so much useful.
`new Polygon(Polygon.getRegularPolygonPoints({ numVertexes, radius }), options);`

is the extra wrapper worth it? I understand left and top equal 0 are part of it, but where the polygon is positioned seems like not something we really care about. I would argue most of people would say that center of the canvas is a better default.
On top of that if i use originX/Y center, defaulting the top left to 0,0 does not really help

Reply:
i would say of not having this extra static method

Reply:
It is worth it.
Setting left, top to 0 fixes the position for all cases of origin. Without it it is fixed at origin center, no matter what origin you pass to it since `getRegularPolygonPoints` is calculated from origin point 0,0 center.
https://github.com/fabricjs/fabric.js/blob/b6f076cc0c817cd7f70abc7028c9d04295d121a8/src/shapes/polyline.class.ts#L73

Reply:
> but where the polygon is positioned seems like not something we really care about

For consistency